tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_all_opcodes_in_container.py

```diff
@@ -129,3 +131,280 @@ def test_all_opcodes_in_container(
+        # On the CALLF opcode we need to reference the second code section, hence the [1] index.
+        opcode_with_data_portion = opcode[0] if opcode != Op.CALLF else opcode[1]
+    else:
+        opcode_with_data_portion = opcode
+
+    bytecode = (Op.PUSH0 * opcode_with_data_portion.min_stack_height) + opcode_with_data_portion
+
+    sections = [Section.Code(code=bytecode)]
+
+    if opcode == Op.CALLF:
+        sections += [Section.Code(code=Op.RETF, code_outputs=0)]
+    elif opcode == Op.EOFCREATE:
+        sections += [
+            Section.Container(
+                container=Container(
+                    sections=[
+                        Section.Code(code=Op.RETURNCONTRACT[0](0, 0)),
+                        Section.Container(Container.Code(code=Op.STOP)),
+                    ]
+                )
+            )
+        ]
+
+    sections += [Section.Data(b"\0" * 32)]
+
+    eof_test(
+        data=Container(
+            sections=sections,
+        ),
+        expect_exception=EOFException.MISSING_STOP_OPCODE,
+    )
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [op for op in list(Op) if op in halting_opcodes],
+)
+def test_all_unreachable_terminating_opcodes_after_stop(
+    eof_test: EOFTestFiller,
+    opcode: Opcode,
+):
+    """
+    Test all terminating opcodes after stop.
+    """
+    match opcode:
+        case Op.STOP:
+            sections = [Section.Code(code=Op.STOP + Op.STOP)]
+        case Op.RETF:
+            sections = [
+                Section.Code(code=Op.CALLF[1] + Op.STOP),
+                Section.Code(code=Op.STOP + Op.RETF, code_outputs=0),
+            ]
+        case Op.JUMPF:
+            sections = [
+                Section.Code(code=Op.STOP + Op.JUMPF[1]),
+                Section.Code(code=Op.STOP),
+            ]
+        case Op.RETURNCONTRACT:
+            sections = [
+                Section.Code(code=Op.EOFCREATE[0](0, 0, 0, 0) + Op.STOP),
+                Section.Container(
+                    container=Container(
+                        sections=[
+                            Section.Code(code=Op.STOP + Op.RETURNCONTRACT[0](0, 0)),
+                            Section.Container(Container.Code(code=Op.STOP)),
+                        ]
+                    )
+                ),
+            ]
+        case Op.RETURN | Op.REVERT | Op.INVALID:
+            sections = [
+                Section.Code(code=Op.PUSH0 + Op.PUSH0 + Op.STOP + opcode),
+            ]
+        case _:
+            raise NotImplementedError(f"Opcode {opcode} is not implemented")
+
+    eof_test(
+        data=Container(
+            sections=sections,
+        ),
+        expect_exception=EOFException.UNREACHABLE_INSTRUCTIONS
+        if opcode != Op.RETURNCONTRACT
+        else EOFException.INCOMPATIBLE_CONTAINER_KIND,
+    )
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [op for op in list(Op) if op in halting_opcodes and op != Op.STOP],
```

Comment:
again for op in list(Op) does not make sense here. you anyway limit it by condition of haltin_opcode.

Reply:
Did a workaround because using a set (`halting_opcodes` in this case) resulted in pytest getting confused in the test cases during collection (because a set is not guaranteed to be iterated in any order). So we now use `sorted`.